lib/utils/compose_ts.ts

```diff
@@ -44,6 +45,44 @@ import type { DeviceInfo } from './device/api';
 import { getBalenaSdk, getChalk, stripIndent } from './lazy';
 import Logger = require('./logger');
 
+/**
+ * Parse build argument input from flags like `--buildArg service1:var1=value1`
+ * to produce an object in a similar format as the build-variables section of
+ * the `./.balena/balena.yml` file, separating service-specific build arguments
+ * from global build arguments.
+ * @param buildargs --buildArg flag input in dictionary format
+ */
+async function parseMetadataFromFlags(
+	buildargs: Dictionary<string>,
+	composition?: Composition,
+): Promise<ParsedBuildArguments> {
+	const serviceNames = Object.keys(composition?.services || {});
+	const ret: ParsedBuildArguments = {
+		global: {},
+		services: {},
+	};
+
+	for (const [argSpec, argValue] of Object.entries(buildargs)) {
+		const i = argSpec.indexOf(':', 1);
+		const service = i > 0 ? argSpec.substring(0, i) : '';
+		const argName = i > 0 ? argSpec.substring(i + 1) : argSpec;
+		if (service && !serviceNames.includes(service)) {
+			throw new ExpectedError(
+				`Cannot find a service with name ${service} specified in build argument: ${argSpec}`,
```

Comment:
@srlowe pointed out that the set of service names to check against (before erring out) should be the union of the services in `docker-compose.yml` and the services in `docker-compose.dev.yml`, in case the user added the latter to override the former:
https://www.flowdock.com/app/rulemotion/i-cli/threads/C9VgXn2zKz8Tdea_a1ShX7Hu3Rp

Reply:
@pdcastro Added in support for also checking `docker-compose.dev.yml` services in my latest commit.